pkg/xds/sync/dataplane_watchdog.go

```diff
@@ -114,6 +114,48 @@ func (d *DataplaneWatchdog) Cleanup() error {
 	}
 }
 
+func (d *DataplaneWatchdog) DataplaneStateExists() bool {
+	if d.dpType == "" {
+		metadata := d.MetadataTracker.Metadata(d.key)
+		if metadata == nil {
+			d.log.Info("[WARNING] could not check state for dataplane without metadata")
+			return false
+		}
+		d.dpType = metadata.GetProxyType()
+	}
+	proxyID := core_xds.FromResourceKey(d.key)
+	switch d.dpType {
```

Comment:
I'd recommend adding some debug-level logs for each case.

Reply:
It's refactored and this code block is removed.